go/vt/tabletserver/query_executor.go

```diff
@@ -295,25 +295,33 @@ func (qre *QueryExecutor) execDDL() (*sqltypes.Result, error) {
 		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "DDL is not understood")
 	}
 
-	conn, err := qre.tsv.te.txPool.LocalBegin(qre.ctx)
+	if qre.transactionID != 0 {
```

Comment:
We should probably fail here instead. DDLs have an implicit commit in them. So, this will mess up our internal transaction state.

The other option is that we end the transaction here with our own commit. This will at least cause future DMLs against the transaction to fail, which is the right behavior.